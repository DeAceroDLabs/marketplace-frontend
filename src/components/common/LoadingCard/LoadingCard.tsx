import Card from "components/common/Card";

interface LoadingCardProps {
  cards: Number;
  variant: "small" | "medium" | "large";
}

const LoadingCard: React.FC<LoadingCardProps> = ({ cards, variant }) => {
  const loadingCards = Array.from(Array(cards).keys()).map((_, index) => {
    return <Card key={index} size={variant} title={" "} loading={true} />;
  });
  return <>{loadingCards}</>;
};

export default LoadingCard;
