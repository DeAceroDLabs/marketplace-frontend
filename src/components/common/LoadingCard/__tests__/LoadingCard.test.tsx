import renderer from "react-test-renderer";
import LoadingCard from "../LoadingCard";

it("renders LoadingCards with no issue", () => {
  const component = renderer.create(<LoadingCard cards={6} variant="small" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
