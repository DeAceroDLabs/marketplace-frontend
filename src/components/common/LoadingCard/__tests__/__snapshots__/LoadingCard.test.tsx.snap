// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoadingCards with no issue 1`] = `
Array [
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="card small loading"
      role="group"
    >
      <div
        className="cardImage"
      />
      <div
        className="cardInfo"
      >
        <div
          className="cardName"
        >
           
           
        </div>
        <div
          className="secondaryInfo"
        >
           
        </div>
      </div>
    </div>
  </div>,
]
`;
